### PR TITLE
khal: build and install man page

### PR DESCRIPTION
**Pull request by dtzWill**
###### Motivation for this change

Man page is useful to have!

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nix-review --run "nix-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

###### Notify maintainers

cc @gebner

### CODE REVIEW

File: pkgs/applications/misc/khal/default.nix [hunk 1]
```diff
@@ -26,11 +26,16 @@ with python3.pkgs; buildPythonApplication rec {
...
-  nativeBuildInputs = [ setuptools_scm ];
+  nativeBuildInputs = [ setuptools_scm sphinx sphinxcontrib_newsfeed ];
   checkInputs = [ pytest ];
 
   postInstall = ''
+    # zsh completion
     install -D misc/__khal $out/share/zsh/site-functions/__khal
+
+    # man page
+    make -C doc man
```
Comment (worldofpeace) [inline pkgs/applications/misc/khal/default.nix]: Makefile doesn't support install for this?

Comment (teto) [reply pkgs/applications/misc/khal/default.nix]: doesn't seem like it https://github.com/pimutils/khal/blob/master/doc/Makefile

**Comment by teto** [comment]
While at it maybe you could install https://github.com/pimutils/khal/blob/master/misc/khal.desktop else I will do it in a follow up PR

**Comment by dtzWill** [comment]
On Mon, 19 Aug 2019 22:07:52 -0700, Matthieu Coudron <notifications@github.com> wrote:
> While at it maybe you could install https://github.com/pimutils/khal/blob/master/misc/khal.desktop else I will do it in a follow up PR

I started, the following addition to postInstall should do the trick:

```
    # desktop
    install -Dm755 misc/khal.desktop -t $out/share/applications
```

This doesn't work as-is however, since the current khal release doesn't
have the `.desktop` file.
Looks like someone we know was behind the PR adding it! ;)

Accordingly I may let you tackle this.  I'll give it a go..

Maybe just fetchpatch the commit (in addition to snippet above),
drop the patch when bumping khal?

**Comment by teto** [comment]
let's wait for the next release then

**Comment by dtzWill** [comment]
Oops, I went ahead and added it :innocent:.  Works here! \o/

**Comment by gebner** [comment]
Thanks.  Works for me.

**Pull request merged**